### PR TITLE
Document GHC-54540 and GHC-93557

### LINKED ISSUE #131: DerivErrDataConsNotAllInScope -> 54540

### PR DESCRIPTION
**Pull request by BinderDavid**
Fixes #131

### CODE REVIEW

File: message-index/messages/GHC-54540/constructorsNotInScope/after/ConstructorsNotInScope.hs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{-# LANGUAGE StandaloneDeriving #-}
+module ConstructorsNotInScope where
+
+import System.IO( Handle )
+
+-- It is not possible to derive an instance for the typeclass in this module.
```
Comment (david-christiansen) [inline message-index/messages/GHC-54540/constructorsNotInScope/after/ConstructorsNotInScope.hs:6]:
```suggestion
-- It is not possible to derive an instance for the type class in this module.
```
Based on the terminology from the original paper (feel free to reject)

File: message-index/messages/GHC-93557/illegalSynonymInstance/index.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+title: Cannot implement a typeclass for type synonyms by default
+---
+
+The programmer defined a data type `RGB`, and a type synonym `T`, and then tried to implement an instance of the typeclass `Eq` for `T`.
```
Comment (david-christiansen) [inline message-index/messages/GHC-93557/illegalSynonymInstance/index.md:5]:
```suggestion
The programmer defined a data type `RGB`, and a type synonym `T`, and then tried to implement an instance of the type class `Eq` for `T`.
```

File: message-index/messages/GHC-93557/index.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+title: Illegal typeclass instance
+summary: Illegal typeclass instance
+severity: error
+introduced: 9.8.1
+---
+
+Using the language version Haskell2010, only certain types are allowed as instances of a typeclass.
+Every typeclass instance in Haskell2010 has the following form:
```
Comment (david-christiansen) [inline message-index/messages/GHC-93557/index.md:9]:
```suggestion
Every type class instance in Haskell2010 has the following form:
```

File: message-index/messages/GHC-93557/index.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+title: Illegal typeclass instance
+summary: Illegal typeclass instance
+severity: error
+introduced: 9.8.1
+---
+
+Using the language version Haskell2010, only certain types are allowed as instances of a typeclass.
```
Comment (david-christiansen) [inline message-index/messages/GHC-93557/index.md:8]:
```suggestion
Using the language version Haskell 2010, only certain types are allowed as instances of a type class.
```

File: message-index/messages/GHC-93557/index.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+Every typeclass instance in Haskell2010 has the following form:
+```
+instance C t where
+````
+Here, `C` is the name of a typeclass, such as `Eq`, `Show` or `Read`, and `t` is a type.
+But only a subset of types `t` is allowed to appear in typeclass instances.
+These types must all have the form `T` or `T a ... b`, where `T` has to be the name of a type introduced by a data or newtype declaration, and the arguments `a` to `b` all have to be type variables.
```
Comment (david-christiansen) [inline message-index/messages/GHC-93557/index.md:15]: I think an example here would be nice - `Maybe a` has `Maybe` for `T`, `Int` has `Int` for `T`, and the type variable `a` doesn't match.

**Review by david-christiansen — COMMENTED** [review]
A weird language nitpick, and one real suggestion

**Comment by BinderDavid** [comment]
Thanks for the comments. I applied the suggested changes in one commit.

**Comment by BinderDavid** [comment]
Ping @david-christiansen . If you don't complain then I will merge this tonight. I think I fixed all your remarks in the way you intended :)

**Comment by david-christiansen** [comment]
Sorry I missed it! No time to look but I trust you

søn. 9. jul. 2023 10.57 skrev David Binder ***@***.***>:

**Pull request merged**